test/Ocelot.AcceptanceTests/ClientRateLimitTests.cs

```diff
@@ -207,6 +208,118 @@ public void should_call_middleware_withWhitelistClient()
+                .Then(x => _steps.ThenRateLimitingHeadersExistInResponse(true))
+                .When(x => _steps.WhenIGetUrlOnTheApiGatewayMultipleTimesForRateLimit("/api/ClientRateLimit", 1))
+                .Then(x => _steps.ThenRateLimitingHeadersExistInResponse(false))
+                .BDDfy();
+        }
+
+        [Fact]
+        public void should_not_set_ratelimiting_headers_on_response_when_DisableRateLimitHeaders_set_to_true()
+        {
+            int port = RandomPortFinder.GetRandomPort();
+
+            var configuration = new FileConfiguration
+            {
+                Routes = new List<FileRoute>
+                {
+                    new()
+                    {
+                        DownstreamPathTemplate = "/api/ClientRateLimit",
+                        DownstreamHostAndPorts = new List<FileHostAndPort>
+                        {
+                            new()
+                            {
+                                Host = "localhost",
+                                Port = port,
+                            },
+                        },
+                        DownstreamScheme = "http",
+                        UpstreamPathTemplate = "/api/ClientRateLimit",
+                        UpstreamHttpMethod = new List<string> { "Get" },
+                        RateLimitOptions = new FileRateLimitRule()
+                        {
+                            EnableRateLimiting = true,
+                            ClientWhitelist = new List<string>(),
+                            Limit = 3,
+                            Period = "1s",
+                            PeriodTimespan = 1000,
+                        },
+                    },
+                },
+                GlobalConfiguration = new FileGlobalConfiguration()
+                {
+                    RateLimitOptions = new FileRateLimitOptions()
+                    {
+                        DisableRateLimitHeaders = true,
+                        QuotaExceededMessage = "",
+                        HttpStatusCode = 428,
+                    },
+                },
+            };
+
+            this.Given(x => x.GivenThereIsAServiceRunningOn($"http://localhost:{port}", "/api/ClientRateLimit"))
+                .And(x => _steps.GivenThereIsAConfiguration(configuration))
+                .And(x => _steps.GivenOcelotIsRunning())
+                .When(x => _steps.WhenIGetUrlOnTheApiGatewayMultipleTimesForRateLimit("/api/ClientRateLimit", 1))
+                .Then(x => _steps.ThenRateLimitingHeadersExistInResponse(false))
+                .When(x => _steps.WhenIGetUrlOnTheApiGatewayMultipleTimesForRateLimit("/api/ClientRateLimit", 2))
+                .Then(x => _steps.ThenRateLimitingHeadersExistInResponse(false))
+                .When(x => _steps.WhenIGetUrlOnTheApiGatewayMultipleTimesForRateLimit("/api/ClientRateLimit", 1))
+                .Then(x => _steps.ThenRateLimitingHeadersExistInResponse(false))
+                .BDDfy();
```

Comment:
This block was copied & pasted from previous method.
We can define a Setup helper to BDDfy.

Reply:
Fixed in commit https://github.com/ThreeMammals/Ocelot/pull/1307/commits/ec15a6551c36119dd16343f9e1959d9d79da9af6